From 7aa728ae021e3b29fb5903ae0ff894a5bd40bbdc Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Tue, 17 Dec 2019 22:23:41 +0100
Subject: [PATCH 1/2] target/i386: Export TAA_NO bit to guests

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <20191217222342.1939034-2-ehabkost@redhat.com>
Patchwork-id: 93164
O-Subject: [RHEL-7.8 qemu-kvm-rhev PATCH 1/2] target/i386: Export TAA_NO bit to guests
Bugzilla: 1779530
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

From: Pawan Gupta <pawan.kumar.gupta@linux.intel.com>

TSX Async Abort (TAA) is a side channel attack on internal buffers in
some Intel processors similar to Microachitectural Data Sampling (MDS).

Some future Intel processors will use the ARCH_CAP_TAA_NO bit in the
IA32_ARCH_CAPABILITIES MSR to report that they are not vulnerable to
TAA. Make this bit available to guests.

Signed-off-by: Pawan Gupta <pawan.kumar.gupta@linux.intel.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 7fac38635e1cc5ebae34eb6530da1009bd5808e4)
Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 target/i386/cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/i386/cpu.c b/target/i386/cpu.c
index 6a1d59c..8d03d0e 100644
--- a/target/i386/cpu.c
+++ b/target/i386/cpu.c
@@ -1148,7 +1148,7 @@ static FeatureWordInfo feature_word_info[FEATURE_WORDS] = {
         .feat_names = {
             "rdctl-no", "ibrs-all", "rsba", "skip-l1dfl-vmentry",
             "ssb-no", "mds-no", NULL, NULL,
-            NULL, NULL, NULL, NULL,
+            "taa-no", NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
             NULL, NULL, NULL, NULL,
-- 
1.8.3.1

